include/abb_librws/rws_rapid.h

```diff
@@ -342,33 +348,147 @@ struct RAPIDRecord : public RAPIDSymbolDataAbstract
 
 protected:
   /**
-   * \brief A method to remove a character from a string and count the number of times it occurred.
-   *
-   * \param input for the string to search.
-   * \param character specifying the character to search for.
+   * \brief The record's type name.
+   */
+  std::string record_type_name_;
+
+  /**
+   * \brief Container for the record's components. I.e. other RAPID records or atomic RAPID data.
+   */
+  std::vector<RAPIDSymbolDataAbstract*> components_;
+};
+
+/**
+ * \brief A struct, for representing the data of a RAPID array of atomic type.
+ */
+template<class T, std::size_t N>
+struct RAPIDArray : public RAPIDSymbolDataAbstract
+{
+public:
+  using value_type = T;
+
+  static_assert(N > 0, "Size template parameter should be greater than 0");
+  static const std::size_t array_size = N;
+
+  /**
+   * \brief Default constructor of a new RAPIDArray object
+   * 
+   */
+  RAPIDArray()
+  {}
+
+  /**
+   * \brief Variadic constructor for a new RAPIDArray object
+   * 
+   * \param values initiale values to fill up the array
+   * 
+   */
+  template<typename... value_type>
+  RAPIDArray(value_type... values)
```

Comment:
Agreed. I had read about it but since it seemed more complicated I went with the simpler solution there. However, indeed it adds an unnecessary overhead. I will change it today.